guides/github.md

```diff
@@ -0,0 +1,130 @@
+supported by GitHub come with usage costs, but are better from the point of
+view of supply chain security than hosting your own infrastructure for CI.
+
+All GitHub Actions workflows should restrict [permissions][gha-permissions] to
+the minimum scope needed. Scope the permissions at job level, instead of at
+workflow level.
+
+```
+  check-tests:
+    permissions:
+      actions: read
+```
+
+Minimize usage of actions that create PRs or push code to branch. Thoroughly
+inspect actions that can approve PRs and workflows that are triggered after a
+PR has been approved (term-of-check-vs-term-of-use type of concerns).
+
+If using actions outside of your organisation, these should be pinned by
+commit hash. Don't pin be version tag as these tags can be forced pushed. The
+[Scorecards Pinned Dependencies workflow][scorecard] can help identify which
+actions should still be pinned. [Dependabot][dependabot] is able to upgrade
+the versions of these actions periodically:
+
+```
+      - name: Upload sdist as an artifact for later jobs in workflow
+        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
+        with:
+          ...
+```
+
+If secrets are used as arguments to commands for the workflows, pass them via
+environment variables. The action log can leak command line arguments!
+
+If using secrets, these should be scrubbed as soon as they are no longer
+needed. Make sure they don't show up in logs, including debug runs of the
+workflow!
+
+Be extra careful of GitHub context variables (e.g.,
+`github.event.issue.title`) which are vulnerable to command injection attacks.
+Don't use these directly in commands. Prefer passing them via environment
+variables. Use [Scorecards][scorecard] to identify dangerous workflows.
+
+Audit carefully all workflows of type `pull_request_target` and
+`workflow_run`. For these, don't use secrets at all and separate the sensitive
+parts into separate runs. Validate all inputs and only run these workflows on
+protected branches. Never use `pull_request_head.ref` as the branch head
+reference since this can be updated between the time the PR gets approved and
+the time the CI workflow starts executing.
+
+### Using self-hosted runners
+
+If using self-hosted runners, create 2 separate pools: one pool to be used
+only for CI for PRs and another one to be used for releases and other
+sensitive jobs. Ideally, don't share these pools across multiple projects in
+the same organisation.
+
+> [!WARNING]
+> Repository-level self-hosted runners are accessible to any workflow in the
+> repository. Never trigger a CI job before the PR is reviewed as a PR can
+> maliciously alter the workflow to attack the hosted runner.
+
+Don't allow any secrets in jobs that run in the CI pool. Only use secrets on
+the release pool and never trigger jobs on the release pool on code that is
+not reviewed.
+
+Ideally, the self-hosted runners should be ephemeral. Destroy them as soon as
```

Comment:
Can you explain how this is supposed to work?

Reply:
One possibility is to have an image of the runner in the cloud account, then use a REST API or similar to trigger creating a new VM based on that image (see https://docs.github.com/en/rest/actions/self-hosted-runners?apiVersion=2022-11-28#create-configuration-for-a-just-in-time-runner-for-an-organization for example). This image stays up for the duration of CI and then gets deleted.

Reply:
I think it would be good to include that link

Reply:
I'll add it to the next PR